12_glob_lab/globlab.txt

```diff
@@ -1,29 +1,30 @@
-Glob Lab.
-https://www.tocode.co.il/bundles/unix1/lessons/globlab
-
-1. Type in a glob pattern to match:
-
-  1. All files whose name starts with a lowercase letter
-  Answer: 
-
-  2. All files whose name starts with an uppercase letter
-  Answer:
-
-  3. All files whose name ends in an "a", "e", "i", "o" or "u"
-  Answer:
-
-  4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
-  Answer:
-
-  5. All files with extension ".c" that are in a folder named "src"
-  Answer:
-
-  6. All files with extension ".c" under any folder
-  Answer:
-
-2. Write a command that would delete all files ending with a ".png" under any subdirectory.
-
-3. The command "ls -l" can take the name of a file or directory and show details on them. From within "wildcards" folder run "ls -l [A-Z]*". What was printed? Why?
-
-4. Search in "man ls" for an option to show only directory data so the command "ls _____ D*" should print only the word Doc
-
+Glob Lab.
+https://www.tocode.co.il/bundles/unix1/lessons/globlab
+
+1. Type in a glob pattern to match:
+
+  1. All files whose name starts with a lowercase letter
+  Answer: ls [a-z]* or ls [[:lower:]]
+
+  2. All files whose name starts with an uppercase letter
+  Answer: ls [A-Z]*????
+       or ls [[:upper:]]
+
+  3. All files whose name ends in an "a", "e", "i", "o" or "u"
+  Answer:ls *[aeiou]
+
+  4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
+  Answer:ls *[!aeiou]
+
+  5. All files with extension ".c" that are in a folder named "src"
+  Answer:ls *.c /src
+
+  6. All files with extension ".c" under any folder
+  Answer: >>>> ls *.c ??????

```

Comment:
This is a continuation of the previous question-

Now that you know that you can use `/` in a glob pattern to separate directories, and `src/*.c` is all c files in src you can replace `src` with star to get:

`ls */*.c` which means all .c files in all folders